src/pages/Lend/AlgoliaPagination.vue

```diff
@@ -1,103 +1,155 @@
 		>
-			<!-- previous page -->
-			<li
-				class="ais-Pagination-item ais-Pagination-item--previousPage"
-				:class="isFirstPage ? 'ais-Pagination-item--disabled' : ''">
-				<span
-					v-if="isFirstPage"
-					aria-label="Previous (Inactive)"
-					class="ais-Pagination-link"
-				>
-					<kv-icon name="triangle" class="disabled" />
-				</span>
-				<a
-					v-if="!isFirstPage"
-					:href="createURL(currentRefinement - 1)"
-					@click.prevent="refine(currentRefinement - 1)"
-					aria-label="Previous"
-					class="ais-Pagination-link"
-				>
-					<kv-icon name="triangle" />
-				</a>
-			</li>
-
-			<!-- pages -->
-			<li
-				v-for="page in pages"
-				:key="page"
-				class="ais-Pagination-item"
-				:class="page === currentRefinement ? 'ais-Pagination-item--selected' : ''"
+			<span
+				v-if="isLastPage"
+				aria-label="Next (Inactive)"
+				class="ais-Pagination-link"
 			>
-				<span
-					v-if="page === currentRefinement"
-					class="ais-Pagination-link"
-				>
-					{{ page + 1 }}
-				</span>
-
-				<a
-					v-if="page !== currentRefinement"
-					:href="createURL(page)"
-					@click.prevent="refine(page)"
-					class="ais-Pagination-link"
-				>
-					{{ page + 1 }}
-				</a>
-			</li>
-
-			<!-- next page -->
-			<li
-				class="ais-Pagination-item ais-Pagination-item--nextPage"
-				:class="isLastPage ? 'ais-Pagination-item--disabled' : ''"
+				<kv-icon name="triangle" class="disabled" />
+			</span>
+
+			<a
+				v-if="!isLastPage"
+				:href="createUrl(currentRefinement + 1)"
+				@click.prevent="refine(currentRefinement + 1)"
+				aria-label="Next"
+				class="ais-Pagination-link"
 			>
-				<span
-					v-if="isLastPage"
-					aria-label="Next (Inactive)"
-					class="ais-Pagination-link"
-				>
-					<kv-icon name="triangle" class="disabled" />
-				</span>
-
-				<a
-					v-if="!isLastPage"
-					:href="createURL(currentRefinement + 1)"
-					@click.prevent="refine(currentRefinement + 1)"
-					aria-label="Next"
-					class="ais-Pagination-link"
-				>
-					<kv-icon name="triangle" />
-				</a>
-			</li>
-		</ul>
-	</ais-pagination>
+				<kv-icon name="triangle" />
+			</a>
+		</li>
+	</ul>
 </template>
 
 <script>
-import { AisPagination } from 'vue-instantsearch';
+import _range from 'lodash/range';
+import _uniq from 'lodash/uniq';
 import KvIcon from '@/components/Kv/KvIcon';
 
 export default {
 	components: {
-		AisPagination,
 		KvIcon,
 	},
 	props: {
-		padding: {
+		currentRefinement: {
 			type: Number,
-			default: 0,
-		}
+			required: true,
+		},
+		createUrl: {
+			type: Function,
+			required: true,
+		},
+		isFirstPage: {
+			type: Boolean,
+			required: true,
+		},
+		isLastPage: {
+			type: Boolean,
+			required: true,
+		},
+		nbPages: {
+			type: Number,
+			required: true,
+		},
+		pages: {
+			type: Array,
+			required: true,
+		},
+		refine: {
+			type: Function,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			maximumExtraPages: 3,
+		};
+	},
+	computed: {
+		numbers() {
```

Comment:
This looks like it was copied directly from KvPagination. Maybe this needs to be done as a mixin?

Reply:
It has some differences so we have another tech-debt ticket to de-dupe the code later

Reply:
👍